### PR TITLE
Improve persisted query performance by caching the document

### PR DESCRIPTION
**Pull request by paulpdaniels**
This PR improves the performance of the `ApolloPersistedQuery` wrapper by caching the parsed `Document` instead of the unparsed string. The advantage of this approach is that in the happy path we get to skip subsequent parsing steps and simply return the `Document`

### CODE REVIEW

File: core/src/main/scala/caliban/wrappers/ApolloPersistedQueries.scala [hunk 1]
```diff
@@ -42,13 +57,12 @@ object ApolloPersistedQueries {
           readHash(request) match {
             case Some(hash) =>
               ZIO
-                .accessM[ApolloPersistence](_.get.get(hash))
+                .serviceWith[Service](_.get(hash))
                 .flatMap {
-                  case Some(query) => UIO(request.copy(query = Some(query)))
-                  case None        =>
+                  case Some(doc) => docVar.set(Some(Right(doc))) as request
```
Comment (frekw) [inline core/src/main/scala/caliban/wrappers/ApolloPersistedQueries.scala:62]: Do we prefer `as x` over `.as(x)`?

**Comment by frekw** [comment]
There is probably a great reason for this that I'm missing, but why are we passing in the `dicVar` rather than just accessing the service directly (since we need to use the service in others code paths anyhow)?

I also think this might be susceptible to DoS attacks by sending ever growing (but correct) queries, should we have some sort of LRU expiry in place (possibly via zio-cache)?

**Comment by frekw** [comment]
Nvm I just grokked the docVar, but I think it needs to be a FiberRef, otherwise parallel requests will overwrite each other's value, no?

File: core/src/main/scala/caliban/wrappers/ApolloPersistedQueries.scala (hunk 1 above)
Comment (paulpdaniels) [reply core/src/main/scala/caliban/wrappers/ApolloPersistedQueries.scala:62]: I'm not sure, I don't think we have a style guide for it, but I'm fine with either I know the infix can get a little out of hand sometimes.

Comment (ghostdogpr) [reply core/src/main/scala/caliban/wrappers/ApolloPersistedQueries.scala:62]: I tend to prefer `.as(x)` and `.` in general except when there's a symbol alias.

**Review by ghostdogpr — APPROVED** [review]

**Comment by frekw** [comment]
Oh right, the `EffectfulWrapper` will create a unique ref per request. My bad! 🙂

I also think we can postpone the DoS stuff and just make a note of it since it's orthogonal to the changes in this PR.

**Review by frekw — APPROVED** [review]

File: core/src/main/scala/caliban/wrappers/ApolloPersistedQueries.scala (hunk 1 above)
Comment (frekw) [reply core/src/main/scala/caliban/wrappers/ApolloPersistedQueries.scala:62]: @ghostdogpr that's usually what I do as well but I don't really have any strong opinions around it so I'm fine with either

**Review by ghostdogpr — APPROVED** [review]
LGTM!

**Pull request merged**